GetIntoTeachingApiTests/Models/GetIntoTeaching/Validators/TeachingEventUpsertOperationValidatorTests.cs

```diff
@@ -40,22 +41,47 @@ public void Validate_WhenExistingEventWithReadableIdHasSameTeachingEventId_HasNo
 
             _mockCrm.Setup(m => m.GetTeachingEvent("existing")).Returns(existingTeachingEvent);
 
-            var result = _validator.Validate(operation);
+            var result = _validator.TestValidate(operation);
 
-            result.IsValid.Should().BeTrue();
+            result.ShouldNotHaveValidationErrorFor(te => te.ReadableId);
         }
 
         [Fact]
-        public void Validate_WhenExistingEventWithReadableIdHasDifferentTeachingEventId_HasErrors()
+        public void Validate_WhenExistingEventWithReadableIdHasDifferentTeachingEventId_HasError()
         {
             var operation = new TeachingEventUpsertOperation() { Id = Guid.NewGuid(), ReadableId = "existing" };
             var teachingEvent = new TeachingEvent() { Id = Guid.NewGuid() };
 
             _mockCrm.Setup(m => m.GetTeachingEvent("existing")).Returns(teachingEvent);
 
-            var result = _validator.Validate(operation);
+            var result = _validator.TestValidate(operation);
 
-            result.IsValid.Should().BeFalse();
+            result.ShouldHaveValidationErrorFor(te => te.ReadableId)
+                .WithErrorMessage("Must be unique");
+        }
+
+        [Theory]
+        [InlineData("eventname", false)]
+        [InlineData("eventname1", false)]
+        [InlineData("event-name", false)]
+        [InlineData("event_name", false)]
+        [InlineData("event name", true)]
+        [InlineData("event?name", true)]
+        [InlineData("event@name", true)]
+        [InlineData("event:name", true)]
+        public void Validate_ReadableId_AllowsOnlyAlphanumericHyphensUnderscores(string readableId, bool hasError)
```

Comment:
I think this will pass with a readable id like `event-name-` though not sure if its a big deal

Reply:
Pushed a change for this. I don't think the Regex is much worse.